helm-charts/splunk-otel-collector/templates/config/_common.tpl

```diff
@@ -80,8 +80,81 @@ resourcedetection:
     # The `system` detector goes last so it can't preclude cloud detectors from setting host/os info.
     # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/resourcedetectionprocessor#ordering
     - system
+  {{- if hasPrefix "gke" (include "splunk-otel-collector.distribution" .) }}
```

Comment:
```suggestion
  {{- if and (hasPrefix "gke" (include "splunk-otel-collector.distribution" .)) .Values.clusterName  }}
```
How about not enabling this attr if clusterName variable is set? The ordering of the processors means that even in eks/gke distros, if clusterName is set, the attr value will be equal to the one set by `resource` processor. So we might as well not detect when clusterName var exists. This would potentially save the extra 5-10s delay with the EKS detector on startup.

Reply:
Good idea, I'll update 👍

Reply:
I ended up just creating a new variable called `autoDetectClusterName`, which allows us to only add the cluster name logic if `clusterName` isn't specified, and the distribution is `eks*` or `gke*`. I tested in `gke` and `kops` with and without `clusterName`, and found no cluster name related config additions (outside of what was already there before this PR) when `clusterName` was specified.